internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -117,19 +128,41 @@ func (ad *ActionDispatcher) Dispatch(ctx context.Context, acker acker.Acker, act
 		strings.Join(detectTypes(actions), ", "),
 	)
 
+	var mErr error
 	for _, action := range actions {
-		if err := ctx.Err(); err != nil {
-			return err
+		if err = ctx.Err(); err != nil {
+			ad.errCh <- err
+			return
 		}
 
 		if err := ad.dispatchAction(ctx, action, acker); err != nil {
+			var e errors.Error
+			if errors.As(err, &e) && e.Type() == errors.TypeRetryableAction {
```

Comment:
Still don't like the usage of custom error type. What about any error results in it being triggered for retry if its retryable?